assets/src/dashboard/components/fileUpload/index.js

```diff
@@ -0,0 +1,301 @@
+  border-width: 2px;
+  background-color: ${({ theme }) => theme.colors.gray25};
+
+  transition: border-color 300ms ease-in;
+`;
+
+const StaticUploadArea = styled.div`
+  position: absolute;
+  top: 0;
+  left: 0;
+  width: 100%;
+  height: 100%;
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  align-items: center;
+  color: ${({ theme }) => theme.colors.gray500};
+  z-index: 0;
+`;
+
+const StaticAreaText = styled.span`
+  ${TypographyPresets.Medium};
+  margin: 0 auto 20px;
+`;
+
+const StaticAreaIcon = styled(_UploadIcon)`
+  width: 52px;
+  height: 52px;
+`;
+const UploadedContentContainer = styled.div`
+  width: 100%;
+  display: flex;
+  flex-direction: row;
+  flex-wrap: wrap;
+  align-items: flex-start;
+  justify-self: flex-start;
+`;
+
+const DeleteButton = styled.button`
+  position: absolute;
+  z-index: 10;
+  border-radius: 100%;
+  width: 30px;
+  height: 30px;
+  margin-top: -10px;
+  margin-left: -10px;
+  background-color: ${({ theme }) => theme.colors.gray75};
+  color: ${({ theme }) => theme.colors.gray500};
+  border: ${({ theme }) => theme.borders.transparent};
+`;
+
+const UploadedContent = styled.div`
+  width: 60px;
+  margin: 0 15px 10px 0;
+
+  ${DeleteButton} {
+    display: none;
+    opacity: 0;
+    transition: opacity 300ms ease-in-out;
+  }
+
+  &:hover,
+  &:focus {
+    ${DeleteButton} {
+      display: block;
+      opacity: 1;
+    }
+  }
+`;
+
+const DisplayImage = styled.img`
+  width: 100%;
+  height: 60px;
+  margin-bottom: 5px;
+  object-fit: cover;
+`;
+
+const DeleteIcon = styled(_DeleteIcon)`
+  width: 100%;
+  height: auto;
+`;
+
+const DisplayTitle = styled.span`
+  ${TypographyPresets.ExtraSmall};
+  display: inline-block;
+  width: 100%;
+  margin-right: 0.5em;
+  white-space: normal;
+  word-break: break-word;
+`;
+
+const UploadLabelAsCta = styled(DefaultButton).attrs({
+  as: 'label',
+})`
+  margin: 0 auto;
+  align-self: flex-end;
+  z-index: 10;
+  font-size: 14px;
+  line-height: 16px;
+`;
+
+function disableDefaults(e) {
+  e.preventDefault();
+  e.stopPropagation();
+}
+
+const FileUpload = ({
+  id,
+  label,
+  onDelete,
+  onSubmit,
+  isFileNameVisible,
+  isMultiple,
+  ariaLabel,
+  emptyDragHelperText = __('You can also drag your file here', 'web-stories'),
+  uploadedContent = [],
+  acceptableFormats = DEFAULT_FILE_UPLOAD_TYPES,
+}) => {
+  const uploadFileContainer = useRef(null);
+  const fileInputRef = useRef(null);
+  const [isDragging, setDragging] = useState(false);
+
+  const handleUploadFile = useCallback(
+    (files) => {
+      onSubmit(Object.values(files));
+    },
+    [onSubmit]
+  );
+
+  const handleChange = useCallback(
+    (event) => {
+      handleUploadFile(event.target.files);
+      fileInputRef.current.value = '';
+    },
+    [handleUploadFile]
+  );
+
+  const handleDragDrop = useCallback(
+    (e) => {
+      disableDefaults(e);
+      const files = e.dataTransfer?.files;
```

Comment:
If this is null, this will be odd to call the `handleUploadFile` callback. We should probably create a `handleUploadError` callback and then manage that.

Reply:
good point. having dataTransfer not be present isn't possible within the event handler for a drop action - I updated this to make more sense so that if somehow files is undefined we just return `isDragging` to false. We'll have error handling via our toaster for when file uploads don't work.